### PR TITLE
Add support for resolving directly circular dependencies on postgres

### PR DESCRIPTION
**Pull request by Page-**
Change-type: minor

### CODE REVIEW

File: src/AbstractSQLCompiler.ts [hunk 1]
```diff
@@ -586,19 +587,46 @@ $$ LANGUAGE ${fnDefinition.language};`);
...
+					const fkDefinition = `FOREIGN KEY ("${fieldName}") REFERENCES "${referencedTable.name}" ("${references.fieldName}")`;
+
+					const schemaInfo = schemaDependencyMap[references.resourceName];
+					if (schemaInfo && schemaInfo.depends.includes(table.resourceName)) {
+						// It's a simple circular dependency so we switch it to an ALTER TABLE
+						// TODO: Remove the postgres specific "IF NOT EXISTS" check
+						alterSchemaStatements.push(`\
+DO $$
+BEGIN
+	IF NOT EXISTS (
```
Comment (roman-mazur) [inline src/AbstractSQLCompiler.ts]: What are the cases this foreign key can already exist?

**Comment by dfunckt** [comment]
Do you have a real need for this one @Page- ? It feels to me it's enabling a use case that is problematic in the first place and should be solved at a higher level.

**Comment by Page-** [comment]
Yes @dfunckt: https://github.com/balena-io/open-balena-api/pull/163

File: src/AbstractSQLCompiler.ts (hunk 1 above)
Comment (Page-) [reply src/AbstractSQLCompiler.ts]: The create schema statements are rerun on pinejs startup to create any new tables that are missing (or in the case of a new env to create everything) so in those cases the foreign keys can already exist and is why all of the create statements include the `IF NOT EXISTS` equivalent

**Pull request merged**